xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/events/AbstractEventStreamEvent.java

```diff
@@ -27,6 +27,12 @@
  */
 public abstract class AbstractEventStreamEvent implements org.xwiki.observation.event.Event
 {
+    /**
+     * Used to provide a key to a property in the current execution context that avoids stepping into a loop when
+     * triggering new events.
+     */
+    public static final String EVENT_LOOP_CONTEXT_LOCK_PROPERTY = "eventLoopContextLockProperty";
```

Comment:
I would keep this internal. By putting it here you are making it a public API (not an *.internal.* package).